File: Data_Structure_using_C_Programming/Queue.c
#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node* next;
};

typedef struct node NODE;
typedef struct node* PNODE;
typedef struct node** PPNODE;

void Enqueue(PPNODE first, int iNo)     // InsertLast
{
    PNODE newn = NULL, temp = NULL;

    newn = (PNODE)malloc(sizeof(NODE));
    newn->data = iNo;
    newn->next = NULL;

    if(NULL == *first)
    {
        *first = newn;
    }
    else
    {
        temp = *first;

        while(temp->next != NULL)
        {
            temp = temp->next;
        }
        
        temp->next = newn;
    }
}

int Dequeue(PPNODE first)     // DeleteFirst
{
    int iValue = 0;
    PNODE temp = NULL;

    if(NULL == *first)
    {
        printf("LinkedList is empty");
        return 0;
    }
    else if((*first)->next == NULL)
    {
        iValue = (*first)->data;
        free(*first);
        *first = NULL;
    }
    else
    {
        temp = *first;
        iValue = (*first)->data;
        *first = temp->next;
        free(temp);
    }
    return iValue;
}

void Display(PNODE first)
{
    while(first != NULL)
    {
        printf("| %d | - ",first->data);
        first = first->next;
    }
    printf("\n");
}

int Count(PNODE first)
{
    int iCount = 0;

    while (first != NULL)
    {
        iCount++;
        first = first->next;
    }

    return iCount;
}

int main()
{
    PNODE head = NULL;
    int iRet = 0;

    Enqueue(&head,11);
    Enqueue(&head,21);
    Enqueue(&head,51);
    Enqueue(&head,101);

    Display(head);

    iRet = Count(head);
    printf("Total elements in Queue is : %d\n",iRet);
    
    iRet = Dequeue(&head);
    printf("removed element is : %d\n",iRet);
    
    iRet = Dequeue(&head);
    printf("removed element is : %d\n",iRet);

    Display(head);

    iRet = Count(head);
    printf("Total elements in Queue is : %d\n",iRet);

    return 0;
}

File: Data_Structure_using_JavaScript_Programming/Queue.js
class node
{
    data
    next

    constructor(no)
    {
        this.data = no
        this.next = null
    }
}

class Queue
{
    #first
    #iCount

    constructor()
    {
        this.#first = null
        this.#iCount = 0
    }

    Enqueue(iNo)           // InsertLast
    {
        let newn = null
        let temp = null

        newn = new node(iNo)

        if(null == this.#first)
        {
            this.#first = newn
        }
        else
        {
            temp = this.#first

            while(temp.next != null)
            {
                temp = temp.next
            }

            temp.next = newn
        }
        this.#iCount++
    }
    
    Dequeue()               // DeleteFirst
    {
        let element = 0

        if(null == this.#first)
        {
            console.log("Stack is empty")
            return
        }
        else
        {
            element = this.#first.data
            this.#first = this.#first.next
            this.#iCount--
            return element
        }
    }

    Display()
    {
        let temp = null
        temp = this.#first

        while(temp != null)
        {
            process.stdout.write("| "+temp.data+" | - ")
            temp = temp.next
        }
        console.log()
    }

    Count()
    {
        return this.#iCount
    }
}

console.log()
console.log("---------------------------------- For Number ----------------------------------------")
console.log()

const qobj1 = new Queue()
let iRet = 0

qobj1.Enqueue(11)
qobj1.Enqueue(21)
qobj1.Enqueue(51)
qobj1.Enqueue(101)
qobj1.Enqueue(111)
qobj1.Enqueue(121)
qobj1.Enqueue(151)

qobj1.Display()

iRet = qobj1.Count()

console.log("Number of nodes in LL is : "+iRet)

iRet = qobj1.Dequeue()

console.log("Removed element is : "+iRet)

iRet = qobj1.Dequeue()

console.log("Removed element is : "+iRet)

qobj1.Display()

iRet = qobj1.Count()

console.log("Number of nodes in LL is : "+iRet)

/////////////////////////////////////////////////////////////////////////////

console.log()
console.log("---------------------------------- For Bigint ----------------------------------------")
console.log()

const qobj2 = new Queue()

qobj2.Enqueue(11874685466666n)
qobj2.Enqueue(21876496896854n)
qobj2.Enqueue(51635465643165n)
qobj2.Enqueue(10164361166635n)
qobj2.Enqueue(11196683565266n)
qobj2.Enqueue(12168163656565n)
qobj2.Enqueue(15136354636565n)

qobj2.Display()

iRet = qobj2.Count()

console.log("Number of nodes in LL is : "+iRet)

iRet = qobj2.Dequeue()

console.log("Removed element is : "+iRet)

iRet = qobj2.Dequeue()

console.log("Removed element is : "+iRet)

qobj2.Display()

iRet = qobj2.Count()

console.log("Number of nodes in LL is : "+iRet)